### ISSUE TITLE
Speed

### ISSUE THREAD

**Issue by mcasto** [opis/database#71]
I have a table with 545 records in a MySQL database. 
I run this code:

```
$st = time();
$rec = $db->from('data_table')->select()->first();
$et = time();

print "Elapsed: " . ($et - $st) . " seconds\n";
```

When I run this code, it takes 2.5+ seconds to return. Every time.

Interestingly, when I put that code into a loop and run it over and over, the first pull takes 2+ seconds and subsequent ones take ~0.6 seconds. To me, that implies there's a caching method involved.

But why does the initial pull of the first record take 2.5+ seconds? That seems ridiculous for pulling one record.

When I run (what I would think is the underlying SQL) `SELECT * FROM data_table LIMIT 1` directly, it takes ~100 ms. So why does Opis take 2.5+ seconds to run that query?

I'm having a similar issue when writing data. Writing/updating those 545 records (inside a transaction) takes ~3.5 minutes.

**Comment by msarca** [comment]
There is a warning in the [documentation](https://docs.opis.io/database/4.x/fetching-records.html) regarding `fetch` and `column`, because we do not use `LIMIT`. 

> You should always filter your records before handling the results of a query, even if you use the first or the column method, otherwise you may encounter performance issues when querying over large data sets.

It's very hard for me to figure out what exactly is the cause for your performance issues, since I don't know a thing about your database setup. We have customers that are using Opis Database to handle hundreds of millions of records, with billions of read/write operations every month without breaking a sweat.

> When I run (what I would think is the underlying SQL) SELECT * FROM data_table LIMIT 1 directly, it takes ~100 ms.

You should really check out your database setup because that number is way to high.

**Comment by mcasto** [comment]
Oh. Yeah. The reason my direct SQL command is high is that I'm actually reaching out to a DB on a remote server ... and, now that you pointed me in that direction of thought, the whole thing is reaching out to a remote DB (since I'm running the script on my local dev machine). I'll have to put the script on the remote machine and see how it performs there.

I should have considered that before posting, but I hadn't been awake very long, so I'll blame brain fog :D 

Thanks for your time.

**Comment by mcasto** [comment]
Wow. I've been using PHP for 20+ years, and doing web dev and general programming longer than that ... but this whole thread was the result of a complete newbie mistake. When I pulled the DB to my dev server and ran the script, that insert/update of 545 records took < 1 second ... as I expected it should.

Sorry I brought this into an issue thread since it was _purely_ a PEBCAK issue. I'm gonna put on my dunce cap and sit in the corner while I think about what I did.

Thanks for the patience :D
